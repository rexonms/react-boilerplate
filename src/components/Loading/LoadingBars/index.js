import React from 'react';

import Styles from './LoadingBars.scss';

const propTypes = {};

const defaultProps = {};

const LoadingBars = props => (
  <div className={Styles.spinner}>
    <div></div>
    <div className={Styles.rect2}></div>
    <div className={Styles.rect3}></div>
    <div className={Styles.rect4}></div>
    <div className={Styles.rect5}></div>
  </div>
);

LoadingBars.defaultProps = defaultProps;
LoadingBars.propTypes = propTypes;

export default LoadingBars;
