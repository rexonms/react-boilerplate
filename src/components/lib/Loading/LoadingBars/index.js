import React from 'react';

import Styles from './LoadingBars.scss';

const propTypes = {};

const defaultProps = {};

const LoadingBars = () => (
  <div className={Styles.spinner}>
    <div />
    <div className={Styles.rect2} />
    <div className={Styles.rect3} />
    <div className={Styles.rect4} />
    <div className={Styles.rect5} />
  </div>
);

LoadingBars.defaultProps = defaultProps;
LoadingBars.propTypes = propTypes;

export default LoadingBars;
