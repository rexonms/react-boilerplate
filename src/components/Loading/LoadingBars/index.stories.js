import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import LoadingBars from './index';

const stories = storiesOf('components/Loading/LoadingBars', module);

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('LoadingBars', withInfo(`
  Component
  ~~~js
    <LoadingBar/>)
  ~~~
`)(() => {
  return (
    <LoadingBars/>
  );
}));
