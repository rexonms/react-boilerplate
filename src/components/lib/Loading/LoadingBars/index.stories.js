import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import LoadingBars from './index';

const stories = storiesOf('components/lib/Loading/LoadingBars', module);

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('LoadingBars', withInfo(`
  Standard Loading bar
  ~~~js
    <LoadingBar />
  ~~~
`)(() => <LoadingBars />));
