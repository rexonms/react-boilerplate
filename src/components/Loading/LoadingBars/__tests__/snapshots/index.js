import React from 'react';
import { shallow } from 'enzyme';

import LoadingBars from '../../index';

// https://hackernoon.com/oh-snap-snapshots-unit-integration-testing-with-jest-enzyme-13cc18aecb7b
it('Renders component correctly', () => {
  const component = shallow(<LoadingBars />);
  expect(component).toMatchSnapshot();
});
