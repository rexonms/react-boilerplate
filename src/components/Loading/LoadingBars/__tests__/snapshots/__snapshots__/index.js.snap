// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingBars />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <div
          className="rect2"
/>,
        <div
          className="rect3"
/>,
        <div
          className="rect4"
/>,
        <div
          className="rect5"
/>,
      ],
      "className": "spinner",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rect2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rect3",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rect4",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rect5",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <div
            className="rect2"
/>,
          <div
            className="rect3"
/>,
          <div
            className="rect4"
/>,
          <div
            className="rect5"
/>,
        ],
        "className": "spinner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rect2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rect3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rect4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rect5",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
